### ISSUE TITLE
AWS.MediaLive.list_channels/1 returns a status_code: 403 response

### ISSUE THREAD

**Issue by pranavraja** [aws-beam/aws-elixir#76]
To reproduce:

```elixir
client = AWS.Client.create(...)
client |> AWS.MediaLive.list_channels()
```

This is because lib/aws/request.ex always sends a `payload`, even for `:get` requests:
```elixir
    payload =
      if send_body_as_binary? do
        Map.fetch!(input, "Body")
      else
        encode!(client, metadata.protocol, input)
      end
```

This causes an empty JSON object `{}` to be sent in the HTTP request body, which gets rejected by the AWS API with 403.
Changing the payload to empty string for `:get` requests fixes the issue.

Happy to raise a PR to fix this, just not sure exactly where in the code would be most suited to add this condition.

**Comment by philss** [comment]
Hi @pranavraja :wave: 

Perhaps we shouldn't encode when the `input` is `nil`. Or maybe GET requests shouldn't be encode. I'm not sure :/
I will take a look later this week, but if you don't mind to confirm that and open the PR, it would be awesome! :D

**Comment by pranavraja** [comment]
Yeah not sure which option would be cleaner, in any case I checked the repo and all uses of `:get` pass in `nil` as the `input` anyway, whereas other http verbs do pass in an `input`, so it seems either way would fix the issue. Do you have a preference?

**Comment by philss** [comment]
@pranavraja sorry for the delay. I think we can go with not encoding `GET` requests. WDYT?

**Comment by vrcca** [comment]
[Checking the API.json for this,](https://github.com/aws/aws-sdk-go/blob/main/models/apis/medialive/2017-10-14/api-2.json#L8266) their shape doesn't include a `body` type, but only `querystring` types. Maybe the generator could check this instead? 🤔
